src/_ert/forward_model_runner/client.py

```diff
@@ -26,107 +22,139 @@ class ClientConnectionClosedOK(Exception):
 
 
 class Client:
-    DEFAULT_MAX_RETRIES = 10
-    DEFAULT_TIMEOUT_MULTIPLIER = 5
-    CONNECTION_TIMEOUT = 60
+    DEFAULT_MAX_RETRIES = 5
+    DEFAULT_ACK_TIMEOUT = 5
+    _receiver_task: asyncio.Task[None] | None
 
     def __enter__(self) -> Self:
+        self.loop.run_until_complete(self.__aenter__())
         return self
 
+    def term(self) -> None:
+        self.socket.close()
+        self.context.term()
+
     def __exit__(self, exc_type: Any, exc_value: Any, exc_traceback: Any) -> None:
-        if self.websocket is not None:
-            self.loop.run_until_complete(self.websocket.close())
+        self.loop.run_until_complete(self.__aexit__(exc_type, exc_value, exc_traceback))
         self.loop.close()
 
-    async def __aenter__(self) -> "Client":
+    async def __aenter__(self) -> Self:
+        await self.connect()
         return self
 
     async def __aexit__(
         self, exc_type: Any, exc_value: Any, exc_traceback: Any
     ) -> None:
-        if self.websocket is not None:
-            await self.websocket.close()
+        try:
+            await self._send("DISCONNECT")
+        except ClientConnectionError:
+            logger.error("No ack for dealer disconnection. Connection is down!")
+        finally:
+            self.socket.disconnect(self.url)
+            await self._term_receiver_task()
+            self.term()
+
+    async def _term_receiver_task(self) -> None:
+        if self._receiver_task and not self._receiver_task.done():
+            self._receiver_task.cancel()
+            await asyncio.gather(self._receiver_task, return_exceptions=True)
```

Comment:
Do we have to gather this cancelled task, or can we await it (and suppress asyncio.cancellationerror if it is raised)?

Reply:
Since we capture `asyncio.CancelledError` we can just do await. :+1